text/0000-fabric-private-chaincode.md

```diff
@@ -0,0 +1,284 @@
+ 
+
+ 
+ 
+ 
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+ 
+## FPC Validation
+ 
+When the peer receives a transaction produced by an FPC chaincode, the FPC Validator fetches the corresponding validation data FPC Registry - specifically, the attestation for the chaincode enclave(s) that executed that transaction. It verifies that the attestation is correct by verifying a signature produced by the TEE vendor, and checks that the attestation contains the chaincode hash (MRENCLAVE) that is referenced in the chaincode definition. If this check succeeds, the public key attached to the attestation is used to verify the FPC endorsement signature. Using the FPC Validator, a peer can check that the read/writeset of a transaction was indeed produced by a proper FPC Chaincode.
+ 
+In addition to the peer-side validation, clients must also be able to verify FPC transactions.
+This process is similar to the validation step as described above; the FPC client SDK extension will implement this functionality and make this process thereby transparent to the users. A low level FPC client SDK API may also provide the functionality to the end user directly.
+
+## TEE platform support
+
+Currently, our FPC Runtime and the SDK focuses on Intel SGX SDK. However, components such as the FPC Registry are already designed to support attestations by other TEE platforms as they mature and gain remote attestation capabilities. Also, other components such as the Go part of the FPC chaincode library don't have an Intel SGX depency and can easily be reused. We plan to explore other TEE platforms such as AMD SEV in the future.
+ 
+## A opportunity for Endorsement Policies
+
+Using FPC enables a new way to write Endorsement Policies. In particular, it enables to create policies which take into account that a chaincode is executed in a protected environment (i.e., TEE). Depending on the workload, this can reduce the number of required endorsements and still protect the integrity of the execution result.
```

Comment:
Please refer to some of my previous comments on the importance of endorsement policy in Fabric.